frontend/src/metabase/dashboard/components/DashboardEmbedAction/DashboardEmbedAction.unit.spec.tsx

```diff
@@ -0,0 +1,135 @@
+import userEvent from "@testing-library/user-event";
+import { renderWithProviders, screen } from "__support__/ui";
+import { createMockDashboard, createMockUser } from "metabase-types/api/mocks";
+import {
+  createMockSettingsState,
+  createMockState,
+} from "metabase-types/store/mocks";
+import { DashboardEmbedAction } from "metabase/dashboard/components/DashboardEmbedAction/DashboardEmbedAction";
+
+interface SetupProps {
+  isAdmin: boolean;
+  hasPublicLink?: boolean;
+  editing?: boolean;
+  embeddingEnabled?: boolean;
+  fullscreen?: boolean;
+  linkEnabled?: boolean;
+  publicLinksEnabled?: boolean;
+}
+
+// Mock embedding modal as we don't need its content for the tests and causes
+// some issues with routing for one or two tests.
+jest.mock(
+  "metabase/dashboard/containers/DashboardSharingEmbeddingModal",
+  () => {
+    return {
+      __esModule: true,
+      default: () => <div data-testid="dashboard-sharing-embedding-modal" />,
+    };
+  },
+);
+
+const setup = ({
+  isAdmin,
+  hasPublicLink,
+  publicLinksEnabled = false,
+}: SetupProps) => {
+  const testDashboard = createMockDashboard({
+    public_uuid: hasPublicLink ? "123" : undefined,
+  });
+
+  const TEST_USER = createMockUser({ is_superuser: isAdmin });
+
+  renderWithProviders(<DashboardEmbedAction dashboard={testDashboard} />, {
+    storeInitialState: createMockState({
+      currentUser: TEST_USER,
+      settings: createMockSettingsState({
+        "enable-public-sharing": publicLinksEnabled,
+      }),
+    }),
+  });
+};
+
+describe("DashboardEmbedHeaderButton", () => {
+  describe("whether the button should be rendered", () => {
+    describe("when the user is admin", () => {
+      it("should render the button even if there is no public link", () => {
```

Comment:
Yeah, the idea is that the tests for `DashboardEmbedHeaderButton` would handle this, and the `DashboardEmbedAction` tests would cover the rendering logic. I didn't want to have everything in one place as it would just be polluted and hard to find relevant tests

Reply:
I added the tests here in addition to the ones for the sub-component

Reply:
I think testing everything would be nice (like what you're adding here). At least we should assert a single path of the subcomponent, but not necessary have to permutate through every possibilities of the subcomponent states since we're already have tests for subcomponents.